autofaiss/external/optimize.py

```diff
@@ -474,20 +472,22 @@ def optimize_and_measure_indices(
     use_gpu: bool,
 ):
     """Optimize a list of indices by selecting the best hyperparameters and calculate their metrics"""
-    fs, _ = fsspec.core.url_to_fs(indices_folder)
+    if index_path is None:
+        index_path = "knn.index"
```

Comment:
why add that default ?
I think if index_path is none the correct thing to do would be to raise an Error

Reply:
Just do not let users fail at this last step.
But I am okay to raise an error.

Reply:
good point, we should probably raise at the very beginning (don't we do that already actually in this case?)

Reply:
Yes, we have some check at the begining, it should have covered this edge case.